modules/database/db_flatfile.cpp

```diff
@@ -349,14 +346,11 @@ class DBFlatFile : public Module, public Pipe
 					this->Write("Unable to write database " + db_name);
 
 					f->close();
-
-					if (Anope::IsFile((db_name + ".tmp").c_str()))
-						rename((db_name + ".tmp").c_str(), db_name.c_str());
 				}
 				else
 				{
 					f->close();
-					unlink((db_name + ".tmp").c_str());
+					rename((db_name + ".tmp").c_str(), db_name.c_str());
```

Comment:
On UNIX this is fine as `rename()` will overwrite the old database but on Windows this will fail and set `errno` to `EACCES`: https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/rename-wrename?view=msvc-170

You probably want to add an `#ifdef _WIN32` block that calls `remove()` on the old database before renaming.

Reply:
Thanks, done!